### PR TITLE
CB-12592 : Refactor and add delete require.cache to cordova_util and create packageManager to load and save files.

### PR DESCRIPTION
**Pull request by audreyso**
### What does this PR do?

Refactor and add delete require.cache to cordova_util and create packageManager to load and save files.

### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: cordova-lib/src/cordova/metadata/ubuntu_parser.js [hunk 1]
```diff
@@ -109,8 +110,8 @@ ubuntu_parser.prototype.update_manifest = function() {
         content += '\nIcon=qmlscene';
         console.warn('missing icon element in config.xml');
     }
-    fs.writeFileSync(path.join(this.path, 'manifest.json'), JSON.stringify(manifest));
-    fs.writeFileSync(path.join(this.path, 'cordova.desktop'), content);
```
Comment (audreyso) [inline cordova-lib/src/cordova/metadata/ubuntu_parser.js]: Would you replace this with packageManager.save or just in cases when JSON.stringify is used??

File: cordova-lib/src/cordova/package_manager.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+
+var fs = require('fs');
+
+// packageManager loads package.json files without require so that there is never a cache
```
Comment (stevengill) [inline cordova-lib/src/cordova/package_manager.js]: it loads any `.json` files, not just `package.json`

File: cordova-lib/src/cordova/platform.js [hunk 3]
```diff
@@ -125,7 +125,7 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                     } else if (pkgJson.dependencies[platform]) {
                         spec = pkgJson.dependencies[platform];
                     }
-                    delete require.cache[require.resolve(path.join(projectRoot, 'package.json'))];
+                    pkgJson = cordova_util.requireNoCache(path.join(projectRoot, 'package.json'));
```
Comment (stevengill) [inline cordova-lib/src/cordova/platform.js]: i don't think this line is needed

File: cordova-lib/src/cordova/platform.js [hunk 4]
```diff
@@ -154,8 +154,9 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 return downloadPlatform(projectRoot, platform, spec, opts);
             }).then(function(platDetails) {
                 if(fs.existsSync(path.join(projectRoot, 'package.json'))) {
-                    delete require.cache[require.resolve(path.join(projectRoot, 'package.json'))];
+                    pkgJson = cordova_util.requireNoCache(path.join(projectRoot, 'package.json'));
```
Comment (stevengill) [inline cordova-lib/src/cordova/platform.js]: I think this line + if statement can be deleted too

File: cordova-lib/src/cordova/platform.js [hunk 5]
```diff
@@ -410,11 +409,10 @@ function getPlatformDetailsFromDir(dir, platformIfKnown){
 
     try {
         var pkgPath = path.join(libDir, 'package.json');
-        delete require.cache[pkgPath];
-        var pkg = require(pkgPath);
+        var pkg = cordova_util.requireNoCache(pkgPath);
         platform = platformFromName(pkg.name);
         version = pkg.version;
-        delete require.cache[pkgPath];
+        cordova_util.requireNoCache(pkgPath);
```
Comment (stevengill) [inline cordova-lib/src/cordova/platform.js]: Don't think this is needed

File: cordova-lib/src/cordova/platform_metadata.js [hunk 6]
```diff
@@ -8,7 +8,7 @@
     with the License.  You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
-
+
```
Comment (stevengill) [inline cordova-lib/src/cordova/platform_metadata.js]: unnecessary tab

**Pull request closed without merging**